### ISSUE TITLE
ANSI files with corner separator don't highlight columns

### ISSUE THREAD

**Issue by favasa** [BdR76/CSVLint#66]
In ANSI files with a corner separator, the program detects columns and validate data, but not highlight columns.

When pressing Reformat, colours appear, but also appear a Â char surrounding the separator.

Sample files and screenshots attached.

[Corner_UTF8.csv](https://github.com/BdR76/CSVLint/files/11394662/Corner_UTF8.csv)
[Corner_ANSI.csv](https://github.com/BdR76/CSVLint/files/11394663/Corner_ANSI.csv)
![ANSI_Before_Reformat](https://user-images.githubusercontent.com/132549271/236137147-737ff9d6-9640-47cd-a382-beea8095b8e5.png)
![ANSI_After_Reformat](https://user-images.githubusercontent.com/132549271/236137150-67c0cd12-7275-429d-9541-57e80f342292.png)
![UTF8](https://user-images.githubusercontent.com/132549271/236137153-bde7080b-4ce0-4c4a-8973-76ad8c7a086d.png)

Notepad++ debug info:

[Notepad++_DebugInfo.txt](https://github.com/BdR76/CSVLint/files/11394726/Notepad%2B%2B_DebugInfo.txt)

**Comment by rdipardo** [comment]
Looks like a duplicate of #52

Try a newer version: https://github.com/BdR76/CSVLint/releases

**Comment by favasa** [comment]
I've tried the last version today. Now the strange char doesn't appear by highlight is not working after formatting:

![image](https://user-images.githubusercontent.com/132549271/236772944-20b36cfb-0229-4036-a5ee-83c7aa12b6a6.png)

**Comment by rdipardo** [comment]
> highlight is not working after formatting

That's a separate issue; nothing to do with the column separator character.

The same thing happens every time a file is reloaded (`File` > `Reload from Disk`, or press <kbd>CTRL</kbd>+<kbd>R</kbd>).

Here's a demonstration where `;` is the column separator:

![reloaded-bufr-not-lexed](https://user-images.githubusercontent.com/59004801/236958008-e350e4d6-58a3-4f43-a4ad-c57fb72d6c12.gif)

**Comment by favasa** [comment]
Ok... Close that issue then. Thanks!

**Comment by BdR76** [comment]
Actually no, this issue is not solved yet in the current version, the example files UTF8 and ANSI are still displayed like in the screenshot below.

![csvlint_corner_separator](https://github.com/BdR76/CSVLint/assets/7793956/74523c83-cbbc-430b-ae09-927719731ff3)

The problem is that the separator character is passed to the Lexer through the [ScintillaGateway.SetProperty](https://github.com/BdR76/CSVLint/blob/master/CSVLintNppPlugin/PluginInfrastructure/ScintillaGateway.cs#L4822) function and recieved by the [Lexer.PropertySet](https://github.com/BdR76/CSVLint/blob/master/CSVLintNppPlugin/PluginInfrastructure/Lexer.cs#L396) function.

It is passed to the SetProperty like `¬` (char 172) but then it is received in PropertySet like a two-character string `Â¬` (char 194 + char 172). and the Lexer just took `value[0]` the first character from the string. So under the hood it incorrectly takes the `Â` as the separator character. This is why the CSV Lint lexer makes the file all one color, just blue, becuase it doesn't find any separator.

I've just now changed this line so it takes the last character:
```
-- old:   separatorChar = value[0];
++ new:   separatorChar = value[value.Length-1]; //hack to fix syntax highlighting for corner ¬ separator
```
You can try it by downloading the [development DLL v0.4.6.5ẞ](https://github.com/BdR76/CSVLint/tree/master/CSVLintNppPlugin/bin), this does give the correct syntax highlighting for both the UTF 8 and ANSI file. But it feels a bit like a hack. The separator character shouldn't be converted to a two-byte value in the first place.

**Comment by BdR76** [comment]
This issue is fixed in the latest version v0.4.6.5, see the [releases page](https://github.com/BdR76/CSVLint/releases). You can download it manually and it will be available in the next Notepad++ update in the Plugin Manager.
